metadata-ingestion/src/datahub/ingestion/source/vertica/common.py

```diff
@@ -0,0 +1,1628 @@
+                    continue
+                self.report.report_entity_scanned(dataset_name, ent_type="projection")
+                if not sql_config.projection_pattern.allowed(dataset_name):
+                    self.report.report_dropped(dataset_name)
+                    continue
+                try:
+                    yield from self._process_projections(
+                        dataset_name, inspector, schema, projection, sql_config, table_tags
+                    )
+                except Exception as e:
+                    logger.warning(
+                        f"Unable to ingest {schema}.{projection} due to an exception.\n {traceback.format_exc()}"
+                    )
+                    self.report.report_warning(
+                        f"{schema}.{projection}", f"Ingestion error: {e}"
+                    )
+                if sql_config.include_projection_lineage:
+
+                    try:
+                        dataset_urn = make_dataset_urn_with_platform_instance(
+                            self.platform,
+                            dataset_name,
+                            self.config.platform_instance,
+                            self.config.env,
+                        )
+
+                        dataset_snapshot = DatasetSnapshot(
+                            urn=dataset_urn,
+                            aspects=[StatusClass(removed=False)],
+                        )
+                        lineage_info = self._get_upstream_lineage_info_Projection(dataset_urn, projection)
+
+                        if lineage_info is not None:
+                            # Emit the lineage work unit
+                            
+                            upstream_lineage = lineage_info
+                            
+                            lineage_mcpw = MetadataChangeProposalWrapper(
+                                entityType="dataset",
+                                changeType=ChangeTypeClass.UPSERT,
+                                entityUrn=dataset_snapshot.urn,
+                                aspectName="upstreamLineage",
+                                aspect=upstream_lineage,
+                            )
+                          
+                           
+                            lineage_wu = MetadataWorkUnit(
+                                id=f"{self.platform}-{lineage_mcpw.entityUrn}-{lineage_mcpw.aspectName}",
+                                mcp=lineage_mcpw,
+                            )
+                           
+                            self.report.report_workunit(lineage_wu)
+                            yield lineage_wu
+                            
+                            
+                            
+
+                    except Exception as e:
+                        logger.warning(
+                            f"Unable to get lieange of Projection {schema}.{projection} due to an exception.\n {traceback.format_exc()}"
```

Comment:
typo: lieange

Reply:
thanks for pointing it out, we have fixed it and the updated code will be in the next commit.